src/Middleware/Diagnostics/test/UnitTests/DeveloperExceptionPageMiddlewareTest.cs

```diff
@@ -117,12 +115,47 @@ public async Task ErrorPageWithoutAcceptHeaderForHtmlReturnsPlainText()
             var response = await server.CreateClient().GetAsync("/path");
 
             // Assert
+            Assert.Equal(StatusCodes.Status500InternalServerError, (int)response.StatusCode);
+
             var responseText = await response.Content.ReadAsStringAsync();
             Assert.Equal("text/plain", response.Content.Headers.ContentType.MediaType);
             Assert.Contains("Test exception", responseText);
             Assert.DoesNotContain("<html", responseText);
         }
 
+        [Fact]
+        public async Task StatusCodeFromBadHttpRequestExceptionIsPreserved()
+        {
+            // Arrange
+            using var host = new HostBuilder()
+                .ConfigureWebHost(webHostBuilder =>
+                {
+                    webHostBuilder
+                    .UseTestServer()
+                    .Configure(app =>
+                    {
+                        app.UseDeveloperExceptionPage();
+                        app.Run(context =>
+                        {
+                            throw new BadHttpRequestException("Not found!", 404);
```

Comment:
404 is awkward because that's actually used elsewhere in the stack. Always test with 418 😁. 
Also use the constant here.